inc/css/class-css-utility.php

```diff
@@ -114,15 +151,75 @@ public function add_item( $params ) {
 			)
 		);
 
+		$params['query'] = $this->get_media_query( $params['query'] );
+
 		if ( ! isset( $this->css_array[ $params['query'] ] ) ) {
 			$this->css_array[ $params['query'] ] = array();
 		}
 
+		$params['properties'] = $this->filter_out_media_queries( $params['selector'], $params['properties'] );
+
 		if ( ! isset( $this->css_array[ $params['query'] ][ $params['selector'] ] ) ) {
-			$this->css_array[ $params['query'] ][ $params['selector'] ] = array();
+			$this->css_array[ $params['query'] ][ $params['selector'] ] = $params['properties'];
+		} else {
+			$this->css_array[ $params['query'] ][ $params['selector'] ] = array_merge(
+				$this->css_array[ $params['query'] ][ $params['selector'] ],
+				$params['properties']
+			);
 		}
+	}
 
-		$this->css_array[ $params['query'] ][ $params['selector'] ] = $params['properties'];
+	/**
+	 * Filter out media query items.
+	 *
+	 * @access public
+	 * @since 2.1.0
+	 * @param array $selector CSS selector.
+	 * @param array $properties CSS properties.
+	 */
+	public function filter_out_media_queries( $selector, $properties ) {
+		$query_items = array_filter(
+			$properties,
+			function( $ar ) {
+				return isset( $ar['media'] );
+			}
+		);
+
+		if ( 0 !== count( $query_items ) ) {
+			if ( ! isset( $this->media_items[ $selector ] ) ) {
+				$this->media_items[ $selector ] = $query_items;
+			} else {
+				$this->media_items[ $selector ] = array_merge( $this->media_items, $query_items );
+			}
+		}
+
+		$properties = array_diff_key( $properties, array_flip( array_keys( $query_items ) ) );
+
+		return $properties;
```

Comment:
you can return this directly

Reply:
Yea but I prefer this way usually, specially for big functions.